onnx_neural_compressor/quantization/tuning.py

```diff
@@ -455,63 +482,85 @@ def autotune(
         tuning_logger.evaluation_start()
-        with tempfile.TemporaryDirectory(prefix="ort.quant.") as tmp_dir:
-            # evaluate API requires str input
-            onnx.save_model(
-                q_model,
-                pathlib.Path(tmp_dir).joinpath(pathlib.Path(model_input).name).as_posix(),
-                save_as_external_data=True,
-                all_tensors_to_one_file=True,
-                location=pathlib.Path(model_input).with_suffix(pathlib.Path(model_input).suffix + "_data").name,
-                size_threshold=1024,
-                convert_attribute=False,
-            )
-            # copy config.json to tmp dir for evaluation, LLMs evaluation may need it
-            if isinstance(model_input, str) and os.path.exists(
-                pathlib.Path(model_input).parent.joinpath("config.json").as_posix()
-            ):
-                import shutil
-
-                shutil.copyfile(
-                    pathlib.Path(model_input).parent.joinpath("config.json").as_posix(),
-                    pathlib.Path(tmp_dir).joinpath("config.json").as_posix(),
-                )
-            eval_result: float = eval_func_wrapper.evaluate(
-                pathlib.Path(tmp_dir).joinpath(pathlib.Path(model_input).name).as_posix()
+        # evaluate API requires str input
+        onnx.save_model(
+            q_model,
+            pathlib.Path(tmp_folder.name).joinpath("eval.onnx").as_posix(),
+            save_as_external_data=True,
+            all_tensors_to_one_file=True,
+            location="eval.onnx_data",
+            size_threshold=1024,
+            convert_attribute=False,
+        )
+        # copy config.json to tmp dir for evaluation, LLMs evaluation may need it
+        if isinstance(model_input, str) and os.path.exists(
+            pathlib.Path(model_input).parent.joinpath("config.json").as_posix()
+        ):
+            shutil.copyfile(
+                pathlib.Path(model_input).parent.joinpath("config.json").as_posix(),
+                pathlib.Path(tmp_folder.name).joinpath("config.json").as_posix(),
             )
+        eval_result: float = eval_func_wrapper.evaluate(pathlib.Path(tmp_folder.name).joinpath("eval.onnx").as_posix())
         tuning_logger.evaluation_end()
         logger.info("Evaluation result: %.4f", eval_result)
         tuning_monitor.add_trial_result(trial_index, eval_result, quant_config)
         tuning_logger.trial_end(trial_index)
         if tuning_monitor.need_stop():
-            best_quant_config: config.BaseConfig = tuning_monitor.get_best_quant_config()
-            best_quant_model = _quantize(
-                model_input, quant_config=best_quant_config, calibration_data_reader=calibration_data_reader
-            )
+            external_data_helper.load_external_data_for_model(q_model, tmp_folder.name)
+            best_quant_model = q_model
             break
+
     tuning_logger.tuning_end()
+    if best_quant_model is None:
```

Comment:
It seems that it won't return the qmodel if none of the qmodels meet the tuning criteria. If so, please update the example to check for existence before saving it.

Reply:
if best_quant_model is None, the program will exit directly and will not run to the line of saving model